mapper/tedge_mapper/src/sm_c8y_mapper/topic.rs

```diff
@@ -0,0 +1,102 @@
+use json_sm::{SoftwareListRequest, SoftwareUpdateRequest};
+use mqtt_client::Topic;
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum SoftwareCommand {
+    SoftwareList,
+    SoftwareUpdate,
+    Cumulocity,
+    UnknownCommand,
```

Comment:
Instead of an `UnknownCommand` enum variant here, I feel this enum should implement the `TryFrom` trait instead of the `From` trait which supports returning an error from the `try_from()` method which you'd want to represent an unsupported topic.